### DISCUSSION TITLE
Is there a way to make the rules "greedy"?

### DISCUSSION THREAD

**Discussion by glyh** [tree-sitter/tree-sitter#1562] [Q&A] [answered]
When a rule conflicts with itself, most of the time I'd like to make it greedy, i.e. consume as much tokens as possible.

Is there a way to achieve this? Thanks a lot! 

I want this feature because sometimes a rule can get too complicated to clarify which precedence will result in the longest match.

**Comment by maxbrunsfeld** [comment, marked as answer]
Often, right precedence will cause the longest match to be preferred. It depends on the nature of the conflict though.

**Reply by glyh** [reply]
Thanks!

**Comment by mrossinek** [comment]
Sorry for bumping this, but is there also another solution to this? I am running into a rather convoluted scenario in which `prec.right` does indeed solve the above problem but causes other issues. When not using `prec.right`, the other issue does not occur but I obtain sibling nodes of identical type which I would like to be merged into single ones (i.e. ensure the maximum length node taking precedence over neighboring shorter nodes).
